### PR TITLE
Warn Users before session expires and give them option to refresh

### PR DESCRIPTION
**Pull request by SegiNyn**

### CODE REVIEW

File: indico/web/flask/session.py [hunk 1]
```diff
@@ -229,6 +229,12 @@ def save_session(self, app, session, response):
             # session contents so we need to add a `Vary: Cookie`` header
             response.vary.add('Cookie')
 
+        _non_session_refreshing_endpoints = {
+            'core.session_expiration'
```
Comment (ThiefMaster) [inline indico/web/flask/session.py:233]: I think this should go into `should_refresh_session` (together with the regex i suggested in the comment on the other PR)

**Comment by SegiNyn** [comment]
Hmmm I thought we wanted to review this PR before it would be merged!

File: indico/web/flask/session.py (hunk 1 above)
Comment (SegiNyn) [reply indico/web/flask/session.py:233]: Okay, I'll add this

**Comment by cbartz** [comment]
Sorry for that, I thought it was worth doing the tests. You can review here and create a follow-up PR, then ping me when it is ready for merging.

**Comment by SegiNyn** [comment]
No Worries :) okay, I'll create another Pr for the changes. Thank you!

**Pull request merged**